src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -201,144 +214,143 @@ protected TupleBatch fetchNextReady() throws IOException, DbException {
-          truncatedQuotedField = true;
+          nextRecordTruncated = true;
         } else {
           throw e;
         }
       }
 
-      if (record != null) {
+      /*
+       * Here, if we are on the last row, we make sure to read the entire row until we either hit a new line or until we
+       * have read the entire file (this is for the case where a single worker might be reading a single large row that
+       * was split among other workers). If we're at the last row and the last worker is reading, we just mark this
+       * final line as finished.
+       */
+      if (onLastRow && !finishedReadingLastRow && !isLastWorker) {
+        long trailingStartByte = partitionEndByteRange + 1;
+        long trailingEndByte = trailingStartByte + byteOverlap - 1;
+        long finalBytePositionFound = trailingStartByte;
+        boolean finalLineFound = false;
 
-        /*
-         * Here, if we are on the last row, we make sure to read the entire row until we either hit a new line or until
-         * we have read the entire file (this is for the case where a single worker might be reading a single large row
-         * that was split among other workers). If we're at the last row and the last worker is reading, we just mark
-         * this final line as finished.
-         */
-        if (onLastRow && !finishedReadingLastRow && !isLastWorker) {
-          long trailingStartByte = partitionEndByteRange + 1;
-          long trailingEndByte = trailingStartByte + byteOverlap - 1;
-          long finalBytePositionFound = trailingStartByte;
-          boolean finalLineFound = false;
-
-          while (!finalLineFound) {
-            /*
-             * If we are within the max byte range, then keep checking for a new line. Otherwise, if we've reached the
-             * end of the file, mark finalLineFound as true.
-             */
-            if (trailingEndByte < maxByteRange) {
-              InputStream trailingEndInputStream =
-                  source.getInputStream(trailingStartByte, trailingEndByte);
-              int dataChar = trailingEndInputStream.read();
-              while (dataChar != -1) {
-                char currentChar = (char) dataChar;
-                if (currentChar == '\n' || currentChar == '\r') {
-                  finalLineFound = true;
-                  break;
-                }
-                dataChar = trailingEndInputStream.read();
-                finalBytePositionFound++;
+        while (!finalLineFound) {
+          /*
+           * If we are within the max byte range, then keep checking for a new line. Otherwise, if we've reached the end
+           * of the file, mark finalLineFound as true.
+           */
+          if (trailingEndByte < maxByteRange) {
+            InputStream trailingEndInputStream =
+                source.getInputStream(trailingStartByte, trailingEndByte);
+            int dataChar = trailingEndInputStream.read();
```

Comment:
Don't we need to increment `finalBytePositionFound` here as well? (I think you could avoid this duplication with a do-while loop.)